greenplumpython/pandas/dataframe.py

```diff
@@ -0,0 +1,235 @@
+from typing import Any, Callable, Dict, Iterable, List, Literal, Optional, Tuple, Union
+
+from sqlalchemy import engine, text
+
+import greenplumpython.dataframe as dataframe
+import greenplumpython.db as db
+import greenplumpython.group as groupby
+import greenplumpython.order as orderby
+
+
+class DataFrame:
+    def __init__(
+        self,
+        data: Union[
+            dataframe.DataFrame, List[Union[Tuple[Any, ...], Dict[str, Any]]], Dict[str, List[Any]]
+        ],
+        columns: Optional[List[str]] = None,
+        con: Optional[db.Database] = None,
+    ) -> None:
+        if isinstance(data, dataframe.DataFrame):
+            self._proxy = data
+        else:
+            assert con is not None
+            if isinstance(data, Dict):
+                self._proxy = con.create_dataframe(columns=data)
+            else:
+                self._proxy = con.create_dataframe(rows=data, column_names=columns)
+
+    def to_sql(
+        self,
+        name: str,
+        con: engine,
+        schema: Union[str, None] = None,
+        if_exists: Literal["fail", "replace", "append"] = "fail",
+        index: bool = False,  # Not Used
+        index_label: Union[str, List[str]] = None,  # Not Used
+        chunksize: Union[int, None] = None,  # Not Used
+        dtype: Union[Dict[str, type], None] = None,  # Not Used
+        method: Literal[None, "multi"] = None,  # Not Used  # type: ignore
+    ) -> int:
+        assert index is False, "DataFrame in GreenplumPython.pandas does not have an index column"
+        assert if_exists in ["fail", "replace"], "Only support if_exists = 'fail' or 'replace'"
+        table_name = name if schema is None else (schema + "." + name)
+        with con.connect() as connexion:  # type: ignore
```

Comment:
connexion -> connection

Reply:
Thx, a little confusion between french and english